### ISSUE TITLE
Rustdoc incorrectly warns about crate-level attributes

### ISSUE THREAD

**Issue by tpwrules** [rust-lang/rust#62184]
If I attempt to `cargo doc` a binary crate that has a crate-level attribute, e.g. `#![no_main]`, then rustdoc will warn (or fail if warnings are denied) that the attribute should be in the root module.

```
$ cargo doc
# ...
 Documenting battmon v0.1.0
error: unused attribute
 --> src/main.rs:3:1
  |
3 | #![no_main]
  | ^^^^^^^^^^^
  |
note: lint level defined here
 --> src/main.rs:2:9
  |
2 | #![deny(warnings)]
  |         ^^^^^^^^
  = note: #[deny(unused_attributes)] implied by #[deny(warnings)]

error: crate-level attribute should be in the root module
 --> src/main.rs:3:1
  |
3 | #![no_main]
  | ^^^^^^^^^^^

error: aborting due to 2 previous errors

error: Could not document `battmon`.
```

Of course, removing the attribute causes the code to fail to build:

```
$ cargo build
# ...
   Compiling battmon v0.1.0
error[E0601]: `main` function not found in crate `battmon`
  |
  = note: the main function must be defined at the crate level but you have one or more functions named 'main' that are not defined at the crate level. Either move the definition or attach the `#[main]` attribute to override this behavior.
note: here is a function named 'main'
 --> src/main.rs:10:1
  |
10| #[rtfm::app(device = crate::hal::target)]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to previous error

For more information about this error, try `rustc --explain E0601`.
error: Could not compile `battmon`.
```

So I have to make sure to add or remove the attribute depending on whether I want to build or document the crate. I could remove `#![deny(warnings)]` but it still makes an ugly and incorrect warning. I would expect rustdoc to compile/examine the code the same way rustc does and avoid triggering this warning.

## Meta
```
rustc 1.36.0-beta.6 (a91f2e56c 2019-06-24)
binary: rustc
commit-hash: a91f2e56c44b7eddaaf75e227cec8925437546e9
commit-date: 2019-06-24
host: x86_64-apple-darwin
release: 1.36.0-beta.6
LLVM version: 8.0
```

Code is attached as a Zip file. You will need to install target `thumbv7em-none-eabihf` to compile it.
[battmon issue.zip](https://github.com/rust-lang/rust/files/3335360/battmon.issue.zip)

**Comment by gendx** [comment]
FYI, I could work around this with the following `cfg_attr` (https://github.com/gendx/tock/commit/1c69c4239da9d15b6490ca9cdbdb1894136d2ab3).

```rust
#![cfg_attr(not(doc), no_main)]
```

**Comment by RefinedSoftwareLLC** [comment]
This error message sent me on a wild goose chase, please fix.

**Comment by lolbinarycat** [comment]
triage: this has been fixed as far as i can tell
